test/integration/ipamperf/cloud.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipamperf
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"sync"
+
+	"github.com/golang/glog"
+
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	"k8s.io/kubernetes/pkg/controller/nodeipam/ipam/cidrset"
+	"k8s.io/kubernetes/test/integration/util"
+)
+
+var (
+	pathInstance  = regexp.MustCompile(`/compute/(v[\d]+|beta|alpha)/projects/([\w|-]+)/zones/([\w|-]+)/instances/([\w|-]+)(\?.*)?`)
+	pathUpdateNet = regexp.MustCompile(`/compute/(v[\d]+|beta|alpha)/projects/([\w|-]+)/zones/([\w|-]+)/instances/([\w|-]+)/updateNetworkInterface(\?.*)?`)
+	pathOperation = regexp.MustCompile(`/compute/(v[\d]+|beta|alpha)/projects/([\w|-]+)/zones/([\w|-]+)/operations/([\w|-]+)(\?.*)?`)
+)
+
+func writeOperationStatus(w io.Writer, selfLink string, header http.Header, statusCode int, status string) error {
+	return json.NewEncoder(w).Encode(&compute.Operation{
+		ServerResponse: googleapi.ServerResponse{
+			Header:         header,
+			HTTPStatusCode: statusCode,
+		},
+		SelfLink: selfLink,
+		Status:   status,
+	})
+}
+
+type cloudServer struct {
+	serveAt      int              // the qps to server at
+	allocateCIDR bool             // whether cloud should allocate CIDR
+	cidrSet      *cidrset.CidrSet // the cidr set if allocateCIDR is true
+
+	lock      sync.Mutex                   // protect access to the instances map
```

Comment:
You can keep it - but the comment not really required (as it's implied from our code convention).